packages/common/src/allo/indexer.ts

```diff
@@ -43,3 +43,85 @@ export async function getCurrentSubgraphBlockNumber(
   );
   return BigInt(res.data._meta.block.number);
 }
+
+/**
+ * Waits until the indexer is synchronized to a specified block number.
+ * @param args - Object with parameters:
+ *  chainId - The chain ID.
+ *  blockNumber - The target block number.
+ *  pollIntervalInMs - Polling interval in milliseconds (default: 1000).
+ * @returns A Promise resolving to the synchronized block number.
+ */
+export const createWaitForIndexerSyncTo = (
+  endpoint: string
+): WaitUntilIndexerSynced => {
+  // Define the function for recursive calls
+  const waitForIndexerSyncTo: WaitUntilIndexerSynced = async (args) => {
+    try {
+      const { chainId, blockNumber, pollIntervalInMs = 1000 } = args;
+
+      if (!endpoint) {
+        throw new Error("Missing endpoint.");
```

Comment:
This error should be thrown one level up, when you create the indexer sync function